### PR TITLE
Wait for Snap accounts to be removed before removing the snap

### PR DESCRIPTION
**Pull request by danroc**
## **Description**

Fixes: https://github.com/MetaMask/accounts-planning/issues/76

This PR updates the `KeyringController` to await account removal, and modifies the UI to wait until all accounts of a snap have been removed before removing the snap (this only affects Accounts Snaps).

The bug linked above was occurring because the snap would be removed before all accounts could be removed from the snap, due to the Keyring Controller missing an `await`.

## **Manual testing steps**

1. Install Flask and the SSK Snap.
2. Create some accounts (3 or more) using the SSK.
3. Remove the SSK Snap.
4. You should **not** see the following error in the logs:

    image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .yarn/patches/@metamask-eth-keyring-controller-npm-13.0.1-06ff83faad.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/dist/KeyringController.js b/dist/KeyringController.js
+index 3644209391f2cdf7ff5e2bd8f94517bff13b0f47..ed63d0ba4cb56df9fe3eb6b0a274f696375dd9dc 100644
+--- a/dist/KeyringController.js
++++ b/dist/KeyringController.js
+@@ -259,7 +259,7 @@ class KeyringController extends events_1.EventEmitter {
+         if (!keyring.removeAccount) {
+             throw new Error(constants_1.KeyringControllerError.UnsupportedRemoveAccount);
+         }
+-        keyring.removeAccount(address);
++        await keyring.removeAccount(address);
```
Comment (gantunesr) [inline .yarn/patches/@metamask-eth-keyring-controller-npm-13.0.1-06ff83faad.patch:10]: Two things we need to take into account when starting patches,

1. Add a comment directly in the patch to explain it's purpose
2. Create a PR to do this same code modification in the respective package and also add the link to the PR or commit to the patch

**Review by gantunesr — CHANGES_REQUESTED** [review]

File: .yarn/patches/@metamask-eth-keyring-controller-npm-13.0.1-06ff83faad.patch (hunk 1 above)
Comment (danroc) [reply .yarn/patches/@metamask-eth-keyring-controller-npm-13.0.1-06ff83faad.patch:17]: Done, although this should be more explicit and applied to other patches as well.

**Comment by plasmacorral** [comment]
Just want to make sure we are on the same page, with this PR in progress are we comfortable moving forward with PR [20998](https://github.com/MetaMask/metamask-extension/pull/20998) as-is?

**Review by gantunesr — APPROVED** [review]
LGTM but some checks are failing

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**